### ISSUE TITLE
Provide Document Serialization/Deserialization Code

### ISSUE THREAD

**Issue by johnoneil** [google/automotive-design-compose#957]
I ran into an issue updating to `v.0.25.0` which wasn't major but probably points out an area for improvement.

Basically we use our own code to serialize/deserialize figma docs, so the introduction of the figma doc header meant we had to update this code.

I thought it might be better if we could have `load` and `save` member to the `SerializedDesignDoc` struct, so we could simply leverage that canonical code to do that work.

Our current code to do loading/saving of docs is getting to be about 75 lines or so, so I think it represents something that should probably be presented as part of the `figma_import` crate. Then things like headers etc could be effectively hidden or more uniformly handled.

Another reason why this would be a good idea would be to improve error handling generally. When running our existing loading code on `v0.25.0` we encountered a generic error that wasn't clear. We now know it's because our code didn't load the Header etc, but at the time there were no immediate clues. The error we encountered is below:

```
running 2 tests
test reducer::reducer_tests::test_reducer_receives_action_renders_2_screens ... memory allocation of 4193132728058642432 bytes failed
```

**Comment by johnoneil** [comment]
Don't know why but the PR isn't associated with this issue.
that's here:
https://github.com/google/automotive-design-compose/pull/1054

**Comment by timothyfroehlich** [comment]
> Don't know why but the PR isn't associated with this issue. that's here: #1054

It'll pick it up if you add something like "Fixes #957" in the issue description. I just manually linked it.
